src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -231,16 +231,12 @@ private FedoraResource getBaseVersion(final FedoraResource resource) {
         // of acl:modes for each particular acl:agent.
         final Map<String, Set<String>> effectiveRoles = new HashMap<>();
         authorizations.stream()
-            .filter(x -> checkAccessTo.test(x) || checkAccessToClass.test(x))
-            .forEach(x -> {
-                Stream.concat(x.getAgents().stream(), dereferenceAgentClasses(x.getAgentClasses()).stream())
-                    .distinct()
-                    .forEach(y -> {
-                        effectiveRoles.putIfAbsent(y, new HashSet<>());
-                        effectiveRoles.get(y).addAll(
-                            x.getModes().stream()
-                                        .map(URI::toString)
-                                        .collect(Collectors.toList()));
+            .filter(checkAccessTo.or(checkAccessToClass))
+            .forEach(auth -> {
+                concat(auth.getAgents().stream(), dereferenceAgentClasses(auth.getAgentClasses()).stream())
+                    .forEach(agent -> {
+                        effectiveRoles.putIfAbsent(agent, new HashSet<>());
```

Comment:
`effectiveRoles.computeIfAbsent(agent, agent -> new HashSet<>).addAll(…)` and remove the braces?